### ISSUE TITLE
Print Sheets Revit 2021

### ISSUE THREAD

**Issue by DYoung-ORW** [eirannejad/pyRevit#887]
Print Sheets does not see any sheets in the current project or linked models in Revit 2021. Revit 2019 works perfectly.

pyRevit 4.7.6
Revit 21.0.0.383

```
==> Registered Clones (full git repos)
==> Registered Clones (deployed from archive/image)
master | Deploy: "basepublic" | Branch: "master" | Version: "4.7.6" | Path: "C:\Users\dyoung\AppData\Roaming\pyRevit-Master"
==> Attachments
master | Product: "2021 First Customer Ship" | Engine: 277 | Path: "C:\Users\dyoung\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\dyoung\AppData\Roaming\Autodesk\Revit\Addins\2021\pyRevit.addin"
master | Product: "2019.2.2 Hotfix" | Engine: 277 | Path: "C:\Users\dyoung\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\dyoung\AppData\Roaming\Autodesk\Revit\Addins\2019\pyRevit.addin"
master | Product: "2017.2.4" | Engine: 277 | Path: "C:\Users\dyoung\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\dyoung\AppData\Roaming\Autodesk\Revit\Addins\2017\pyRevit.addin"
==> Installed Extensions
==> Default Extension Search Path
C:\Users\dyoung\AppData\Roaming\pyRevit\Extensions
==> Extension Search Paths
==> Extension Sources - Default
https://github.com/eirannejad/pyRevit/raw/master/extensions/extensions.json
==> Extension Sources - Additional
==> Installed Revits
2021 First Customer Ship | Version: 21.0.0.383 | Build: 20200220_1100(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2021"
2019.2.2 Hotfix | Version: 19.2.20.24 | Build: 20190808_0900(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2019"
2017.2.4 | Version: 17.0.1158.0 | Build: 20181011_1645(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2017"
==> Running Revit Instances
PID: 18952 | 2021 First Customer Ship | Version: 21.0.0.383 | Build: 20200220_1100(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2021"
PID: 2752 | 2019.2.2 Hotfix | Version: 19.2.20.24 | Build: 20190808_0900(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2019"
==> User Environment
Microsoft Windows 10 [Version 10.0.18363]
Executing User: ORW\dyoung
Active User:
Admin Access: No
%APPDATA%: "C:\Users\dyoung\AppData\Roaming"
Latest Installed .Net Framework: 4.8
No .Net Target Packs are installed.
No .Ne-Core Target Packs are installed.
pyRevit CLI 0.25.0.0
```

![pyRevit](https://user-images.githubusercontent.com/64432603/80405032-e6ec8480-8876-11ea-87c3-2d206bde34c0.png)

**Comment by eirannejad** [comment]
Okay seems like Revit 2021 schedule export encoding has changed from UTF-16 to UTF-8. Fixed this for the next release. You can replace `extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/Print Sheets.pushbutton/script.py` with the file below for now to make it work under 2021

[script.py.zip](https://github.com/eirannejad/pyRevit/files/4565875/script.py.zip)

**Comment by DYoung-ORW** [comment]
That did the trick. Thank you for your help and this amazing tool!

**Comment by sharkmullet** [comment]
for anyone looking for the location, it was here for me:
C:\Users\myname\AppData\Roaming\pyRevit-Master\extensions\pyRevitTools.extension\pyRevit.tab\Drawing Set.panel\Print Sheets.pushbutton

**Comment by thazell** [comment]
Just ran into this issue and for depending on the version of pyrevit you installed you might need to also update this file. 
"pyrevitlib\pyrevit\coreutils\__init__.py"
[update_files.zip](https://github.com/eirannejad/pyRevit/files/6756641/update_files.zip)

**Comment by gbalaka** [comment]
**Is there a similar fix for Revit 2022.1. I'm having the same issue. 

I tried replacing the script.py file, but i get an error message when i click on "Print Sheets"**


IronPython Traceback:
Traceback (most recent call last):
 File "C:\Users\galaka\AppData\Roaming\pyRevit-Master\extensions\pyRevitTools.extension\pyRevit.tab\Drawing Set.panel\Print Sheets.pushbutton\script.py", line 1181, in <module>
 File "C:\Users\galaka\AppData\Roaming\pyRevit-Master\extensions\pyRevitTools.extension\pyRevit.tab\Drawing Set.panel\Print Sheets.pushbutton\script.py", line 432, in __init__
 File "C:\Users\galaka\AppData\Roaming\pyRevit-Master\pyrevitlib\pyrevit\forms\__init__.py", line 152, in __init__
 File "C:\Users\galaka\AppData\Roaming\pyRevit-Master\pyrevitlib\pyrevit\forms\__init__.py", line 165, in load_xaml
Exception: Set property 'Microsoft.Scripting.Runtime.DynamicXamlReader+DynamicWriter.SelectionChanged' threw an exception.


Script Executor Traceback:
System.Xaml.XamlObjectWriterException: Set property 'Microsoft.Scripting.Runtime.DynamicXamlReader+DynamicWriter.SelectionChanged' threw an exception. ---> System.MissingMemberException: 'Window_29$30' object has no attribute 'printers_changed'
 at CallSite.Target(Closure , CallSite , Object )
 at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
 at CallSite.Target(Closure , CallSite , Object )
 at Microsoft.Scripting.Runtime.DynamicXamlReader.DynamicWriter.DynamicEventInvoker.SetValue(Object instance, Object value)
 at MS.Internal.Xaml.Runtime.ClrObjectRuntime.SetValue(Object inst, XamlMember property, Object value)
 --- End of inner exception stack trace ---
 at MS.Internal.Xaml.Runtime.ClrObjectRuntime.SetValue(Object inst, XamlMember property, Object value)
 at System.Xaml.XamlObjectWriter.Logic_ApplyPropertyValue(ObjectWriterContext ctx, XamlMember prop, Object value, Boolean onParent)
 at System.Xaml.XamlObjectWriter.Logic_DoAssignmentToParentProperty(ObjectWriterContext ctx)
 at System.Xaml.XamlObjectWriter.WriteEndMember()
 at Microsoft.Scripting.Runtime.DynamicXamlReader.LoadComponent(Object scope, DynamicOperations operations, XamlXmlReader reader)
 at Microsoft.Scripting.Runtime.DynamicXamlReader.LoadComponent(Object scope, DynamicOperations operations, String filename, XamlSchemaContext schemaContext)
 at Microsoft.Scripting.Interpreter.FuncCallInstruction`4.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run5[T0,T1,T2,T3,T4,TRet](T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
 at IronPython.Compiler.Ast.CallExpression.Invoke2Instruction.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run6[T0,T1,T2,T3,T4,T5,TRet](T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5)
 at Microsoft.Scripting.Interpreter.FuncCallInstruction`8.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run8[T0,T1,T2,T3,T4,T5,T6,T7,TRet](T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7)
 at Microsoft.Scripting.Interpreter.DynamicInstruction`8.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run7[T0,T1,T2,T3,T4,T5,T6,TRet](T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6)
 at Microsoft.Scripting.Interpreter.DynamicInstruction`7.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run6[T0,T1,T2,T3,T4,T5,TRet](T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5)
 at IronPython.Runtime.FunctionCaller`2.Default3Call2(CallSite site, CodeContext context, Object func, T0 arg0, T1 arg1)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
 at Microsoft.Scripting.Interpreter.DynamicInstruction`5.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run3[T0,T1,T2,TRet](T0 arg0, T1 arg1, T2 arg2)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
 at IronPython.Runtime.Types.LateBoundInitBinder.FastInitSite`1.CallTarget(CallSite site, CodeContext context, Object inst, T0 arg0)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
 at IronPython.Runtime.Types.PythonType.FastTypeSite`1.CallTarget(CallSite site, CodeContext context, Object type, T0 arg0)
 at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
 at Microsoft.Scripting.Interpreter.DynamicInstruction`4.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.Interpreter.Run(InterpretedFrame frame)
 at Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet](T0 arg0, T1 arg1)
 at IronPython.Compiler.PythonScriptCode.RunWorker(CodeContext ctx)
 at PyRevitLabs.PyRevit.Runtime.IronPythonEngine.Execute(ScriptRuntime& runtime)
